### PR TITLE
Fix broken test

### PR DESCRIPTION
**Pull request by dougyd92**
This test started failing at some point when we added additional URL params to the iframe. I changed it to check the start of the URL, not a strict match.

Other parts of the test also broke with the latest version of Discourse. Some test helpers that used to be available automatically now need to be explicitly imported. There were also intermittent test failures with variables being null or undefined. I think this only happens in the test environment due to race conditions of multiple tests, as they always went away when I ran just a single test case at a time. Anyway, I added some checks for null values.

Related JIRA ticket: https://codecademy.atlassian.net/browse/DISC-617

### Testing instructions
1. Pull the latest code for the discourse repo (Discourse itself, not the plugin).
2. Run the discourse backend `rails s`
3. Run the discourse frontend `bin/ember-cli`
4. To run the automated tests, visit http://localhost:4200/tests?qunit_single_plugin=CodeBytes&qunit_skip_core=1. Verify that everything passes.
5. Also perform manual tests to confirm that the non-test code changes didn't break anything:
6. Visit http://localhost:4200/
7. Create a new topic
8. Click the "Create a Codebyte" button. Verify that valid markdown syntax is inserted and a codebyte appears in the preview window.
9. Select a language. Verify that the markdown in the editor pane is updated.
10. Make some changes in the codebyte in the preview pane and click the "Save to post" button. Verify that the markdown in the editor pane is updated.

### CODE REVIEW

**Comment by christian-dinh** [comment]
apparently the new path is `/tests` and they never told anyone about that until someone asked 💩 

https://meta.discourse.org/t/beginner-s-guide-to-creating-discourse-plugins-part-6-acceptance-tests/32619/35

**Comment by dougyd92** [comment]
> apparently the new path is `/tests` and they never told anyone about that until someone asked 💩
> 
> https://meta.discourse.org/t/beginner-s-guide-to-creating-discourse-plugins-part-6-acceptance-tests/32619/35

Thank you!!! Really annoyed that Discourse keeps changing and breaking things without notice (which is how we got into this mess of the broken button in the first place).

Updated the test plan. Would you mind giving this another look?

File: test/javascripts/acceptance/codebytes-test.js [hunk 1]
```diff
@@ -44,9 +46,8 @@ acceptance("CodeBytes", function (needs) {
       '[codebyte]\n\n[/codebyte]'
     );
 
-    assert.equal(
-      queryAll(".d-editor-preview .d-codebyte iframe").attr('src'),
-      "https://www.codecademy.com/codebyte-editor?lang=&text=",
+    assert.ok(
+      queryAll(".d-editor-preview .d-codebyte iframe").attr('src').startsWith("https://www.codecademy.com/codebyte-editor"),
```
Comment (christian-dinh) [inline test/javascripts/acceptance/codebytes-test.js:50]: might be a bit more semantic to do an `assert.match` with a regex here, assuming that works the same

**Review by christian-dinh — APPROVED** [review]
I don't have a way to run these locally right now but if they're passing for you the code looks good to me!

**Review by zippyzow — APPROVED** [review]
Tests are passing for me and I am able to create and edit a Codebyte.

File: test/javascripts/acceptance/codebytes-test.js (hunk 1 above)
Comment (dougyd92) [reply test/javascripts/acceptance/codebytes-test.js:50]: I looked into your suggestion but unfortunately this is `QUnit.assert` which does not have a `match` function :(

Comment (christian-dinh) [reply test/javascripts/acceptance/codebytes-test.js:50]: woops totally forgot it was qunit not jest, sorry bout that!
